storage/src/mmr/bitmap.rs

```diff
@@ -77,19 +96,98 @@ impl<H: CHasher, const N: usize> Bitmap<H, N> {
             bitmap,
             next_bit: 0,
             mmr: Mmr::new(),
+            pruned_chunks: 0,
         }
     }
 
-    /// Return the number of bitmap chunks that have been pruned.
-    #[inline]
-    fn pruned_chunks(&self) -> usize {
-        self.mmr.pruned_to_pos() as usize
+    /// Initialize a fully pruned bitmap from the metadata in the given partition.
+    ///
+    /// The metadata must store the number of pruned chunks and the pinned hashes corresponding to
+    /// that pruning boundary.
+    pub async fn init_pruned<C: RStorage + Metrics + Clock>(
+        context: C,
+        partition: String,
+    ) -> Result<Self, Error> {
+        let metadata_cfg = MConfig { partition };
+        let metadata = Metadata::init(context.with_label("bitmap_metadata"), metadata_cfg).await?;
```

Comment:
It feels weird to me to have this `metadata` usage on init and write for an "in-memory" implementation. Still mulling this over (and pondering whether there might be a different way to organize).

Reply:
I suppose we could have the user manage the persistence of the data necessary for restoration, but then it feels like we're exposing too much of the implementation?

Reply:
Yeah, I'm not sure what the right answer is. Seems very dumb to push the handling of persistence to someone else.

Reply:
At the risk of sounding "dumb", I suppose the reason this feels weird is that the other trees either write continuously (and have a long-lived db) or don't write at all. This one feels like somewhere in the middle (and for consistency would've expected one of the two existing patterns).

Reply:
I see what you mean. Would you prefer if we just accepted the partition config up front and always wrote the journal each time the pruning boundary changes?  I guess I assumed there might be use cases where you'd never be concerned with persistence at all.